sdk/storage/Azure.Storage.Blobs/tests/ClientSideEncryptionTests.cs

```diff
@@ -0,0 +1,388 @@
+        public async Task Track1DownloadTrack2Blob()
+        {
+            var data = GetRandomBuffer(Constants.KB); // ensure we have enough room in original data
+            var mockKey = new MockKeyEncryptionKey();
+            await using (var disposable = await GetTestContainerEncryptionAsync(
+                new ClientSideEncryptionOptions(ClientSideEncryptionVersion.V1_0)
+                {
+                    KeyEncryptionKey = mockKey,
+                    KeyResolver = mockKey,
+                    KeyWrapAlgorithm = ThrowawayAlgorithmName
+                }))
+            {
+                var track2Blob = disposable.Container.GetBlobClient(GetNewBlobName());
+
+                // upload with track 2
+                await track2Blob.UploadAsync(new MemoryStream(data));
+
+                // download with track 1
+                var creds = GetNewSharedKeyCredentials();
+                var track1Blob = new Microsoft.Azure.Storage.Blob.CloudBlockBlob(
+                    track2Blob.Uri,
+                    new Microsoft.Azure.Storage.Auth.StorageCredentials(creds.AccountName, creds.GetAccountKey()));
+                var downloadData = new byte[data.Length];
+                await track1Blob.DownloadToByteArrayAsync(downloadData, 0, default,
+                    new Microsoft.Azure.Storage.Blob.BlobRequestOptions()
+                    {
+                        EncryptionPolicy = new Microsoft.Azure.Storage.Blob.BlobEncryptionPolicy(mockKey, mockKey)
+                    },
+                    default, default);
+
+                // compare original data to downloaded data
+                Assert.AreEqual(data, downloadData);
+            }
+        }
+
+        [Test]
+        [LiveOnly] // need access to keyvault service && cannot seed content encryption key
+        public async Task RoundtripWithKeyvaultProvider()
+        {
+            var data = GetRandomBuffer(Constants.KB);
+            IKeyEncryptionKey key = await GetKeyvaultIKeyEncryptionKey();
+            await using (var disposable = await GetTestContainerEncryptionAsync(
+                new ClientSideEncryptionOptions(ClientSideEncryptionVersion.V1_0)
+                {
+                    KeyEncryptionKey = key,
+                    KeyWrapAlgorithm = "RSA-OAEP-256"
+                }))
+            {
+                var blob = disposable.Container.GetBlobClient(GetNewBlobName());
+
+                await blob.UploadAsync(new MemoryStream(data));
+
+                var downloadStream = new MemoryStream();
+                await blob.DownloadToAsync(downloadStream);
+
+                Assert.AreEqual(data, downloadStream.ToArray());
+            }
+        }
+
+        [Test]
+        [LiveOnly] // cannot seed content encryption key
+        [Ignore("stress test")]
+        public async Task StressAsync()
+        {
+            static async Task<byte[]> RoundTripDataHelper(BlobClient client, byte[] data)
```

Comment:
nit: maybe RoundTirpDataVerifier?

Suffixes like util, helper don't bring much extra understanding.
https://softwareengineering.stackexchange.com/questions/247267/what-is-a-helper-is-it-a-design-pattern-is-it-an-algorithm